### PR TITLE
Warn when unitless values assigned to unit-requiring fields.  Fixes #1873.


### LINKED ISSUE #1873: Remove functionality that appends 'px' to unitless numbers in STYLE prop
It is dangerous to assume a number that has a distance unit omitted from it, should be pixel-based. 

Working within the [SVG user space](http://www.w3.org/TR/SVG/coords.html#SVGInitialUserCoordinateSystem), it is sometimes required that [number values are left unitless](http://www.w3.org/TR/SVG/coords.html#LengthsInSVG), so that they're proportionate to the user-defined matrix.

The is also the additional overhead in maintaining a property (that accepts a unitless number value) [whitelist](https://github.com/facebook/react/blob/a8fc3b940dbe20c0c7decd19b028215b236c50a6/src/browser/ui/dom/CSSProperty.js#L24), instead of just allowing a pass-through.

### PR DESCRIPTION
**Pull request by jimfb**
Warn when unitless values assigned to unit-requiring fields.  Fixes #1873.

### CODE REVIEW

File: src/browser/ui/dom/dangerousStyleValue.js [hunk 1]
```diff
@@ -13,8 +13,12 @@
 'use strict';
 
 var CSSProperty = require('CSSProperty');
+var ReactCurrentOwner = require('ReactCurrentOwner');
 
 var isUnitlessNumber = CSSProperty.isUnitlessNumber;
+var warning = require('warning');
+
+loggedTypeFailures = {};
```
Comment (sebmarkbage) [inline src/browser/ui/dom/dangerousStyleValue.js]: Missing `var`.

**Comment by sebmarkbage** [comment]
Failing lint but I also don't think we have consensus on whether or not we'll actually want to do this since it's not cross-platform. Let's talk with @vjeux first.

**Comment by jsfb** [comment]
Ok, fixed linted whitespace.  I'll sync up with @vjeux soon, probably after reactconf.

**Comment by ianobermiller** [comment]
@sebmarkbage what are the cross-platform issues here? Do we still plan to deprecate automatically adding `'px'` as per #1873?

**Comment by jimfb** [comment]
@ianobermiller The cross-platform issue is that if you don't automatically add 'px' to the style, then you can't re-use the same style declaration for React components and React-native components, since the numbers in ios aren't intended to be pixels.  Personally, I think it's a questionable idea, because it doesn't seem like you should specify a value that is intended to be pixels on web interpreted differently on mobile.  At best, it feels wacky/hacky, at worst I'm surprised it works at all.  But that's the discussion, obviously there are other points of view.

Personally, I would really like to see https://github.com/facebook/react/issues/1873 fixed.  I feel like anyone who feels strongly about style sharing can pass their unitless style object through a function to apply the correct units for their platform (like Facebook does for css class minification on the web) before passing the object to React, but that's not something React should be doing automatically (and most certainly not in the core).

@sebmarkbage Let's reopen the dialog with @vjeux after Europe.

**Comment by gaearon** [comment]
Let’s keep tracking this in #1873. It does not appear that this particular PR is going to get merged really soon, so I’ll close it as part of the effort to clean up stale PRs.

We can revisit it at any time.

**Pull request closed without merging**